sql/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```
@@ -620,8 +620,17 @@ object SparkConnectClient {
      * Configure the builder using the env SPARK_REMOTE environment variable.
      */
     def loadFromEnvironment(): Builder = {
+      lazy val isAPIModeConnect = Option(System.getProperty("spark.api.mode"))
```

We should throw an error iif api mode is not connect.